### PR TITLE
qemu.tests.single_driver_install: workaround for windows system disk …

### PR DESCRIPTION
**Pull request by CongLi**
…driver change problem.

do a workaround to avoid chaning system disk driver between virtio_blk and virtio_scsi, which will cause BSOD "INACCESSABLE_DEVICE" for lastest windows guests.

Signed-off-by: Cong Li <coli@redhat.com>

id: 1357507

### CODE REVIEW

**Comment by CongLi** [comment]
Guest can boot up successfully after installing virtio_blk driver.

19:07:51 INFO | PASS 4-smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win2012.x86_64.r2.io-github-autotest-qemu.single_driver_install.with_viostor
19:10:53 INFO | PASS 5-smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win2012.x86_64.r2.io-github-autotest-qemu.single_driver_install.with_vioserial
19:13:55 INFO | PASS 6-smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win2012.x86_64.r2.io-github-autotest-qemu.single_driver_install.with_balloon

File: qemu/tests/cfg/single_driver_install.cfg [hunk 1]
```diff
@@ -26,6 +26,10 @@
                 backend_type = passthrough
                 filename_passthrough = /dev/random
         - with_viostor:
+            # For latest windows guests, changing system disk driver between
+            # virtio_blk and virt_scsi will cause BSOD "INACCESSABLE_DEVICE".
+            # Do a workaround here to avoid such scenario.
+            no virtio_scsi
             driver_name = viostor
             device_name = "Red Hat VirtIO SCSI controller"
             drive_format_image1 = ide
```
Comment (suqinhuang) [inline qemu/tests/cfg/single_driver_install.cfg]: 1. only virtio_blk maybe more clear?
2. suggest to remove "drive_format_image1 = ide"
3. we can add variants
e.g
- with_viostor:
    - virtio_blk:
    - virtio_scsi:
     driver_name = vioscsi
     device_name = "Red Hat VirtIO SCSI pass-through controller"

the same as with_vioscsi

Comment (CongLi) [inline qemu/tests/cfg/single_driver_install.cfg]: 1. I think no virtio_scsi is better than only virtio_blk, as there are other drive formats.
2. Have confirmed and test by myself, changing system disk between ide and virtio_blk/virtio_scsi won't cause any problem.
Removing "drive_format_image1 = ide" means virtio_blk / virtio_scsi has been installed in unattended_install successfully.
3. Sorry, I have not got your meaning at point 3.

**Comment by suqinhuang** [comment]
Check with coli,

It's ok for me now.

Acked-by: Suqin Huang

**Comment by yanan-fu** [comment]
LGTM.
Acked-by: Yanan Fu  yfu@redhat.com

**Comment by CongLi** [comment]
Thanks @suqinhuang @yanan-fu , merged.

**Pull request merged**